src/CommunityToolkit.Maui.Core/Essentials/SpeechToText/SpeechToTextImplementation.ios.cs

```diff
@@ -58,27 +55,29 @@ async Task<string> InternalListenAsync(CultureInfo culture, IProgress<string>? r
 				{
 					currentIndex = 0;
 					StopRecording();
+					OnRecognitionResultCompleted(result.BestTranscription.FormattedString);
 					getRecognitionTaskCompletionSource.TrySetResult(result.BestTranscription.FormattedString);
 				}
 				else
 				{
+					if (currentIndex <= 0)
+					{
+						OnSpeechToTextStateChanged(CurrentState);
+					}
+
 					for (var i = currentIndex; i < result.BestTranscription.Segments.Length; i++)
 					{
 						var s = result.BestTranscription.Segments[i].Substring;
 						currentIndex++;
-						recognitionResult?.Report(s);
+						recognitionProgress?.Report(s);
+						OnRecognitionResultUpdated(s);
 					}
 				}
 			}
 		});
 
-		await using (cancellationToken.Register(() =>
-		{
-			StopRecording();
-			getRecognitionTaskCompletionSource.TrySetCanceled();
-		}))
-		{
-			return await getRecognitionTaskCompletionSource.Task;
-		}
+		cancellationToken.ThrowIfCancellationRequested();
+
+		return Task.CompletedTask;
```

Comment:
Completing this method synchrony could could be related to this [issue](https://github.com/CommunityToolkit/Maui/issues/1415)

Reply:
What do you think about merging this PR and addressing https://github.com/CommunityToolkit/Maui/issues/1415 in a future PR? 

As long as this PR isn't introducing that bug (or making it worse), I'd like to merge this because  this PR is currently blocking us from releasing CommunityToolkit.Maui v6.0.0.

Reply:
I'm fine with that, just want to point out if completing in a sync way is the cause of that issue. No problem to merge as it's